packages/create-block/index.ts

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env node
+const fs = require('fs');
+const prompts = require('prompts');
+const path = require('path');
+const spawn = require('cross-spawn');
+const commandLineArgs = require('command-line-args');
+const commandLineUsage = require('command-line-usage');
+
+type Options = {
+  name: string;
+  alias?: string;
+  type: StringConstructor | BooleanConstructor;
+  defaultValue?: string;
+  defaultOption?: string;
+  description: string;
+};
+
+type LanguageType = 'typescript' | 'javascript';
+
+/**
+ * Define the command line options.
+ */
+const options: Options[] = [
+  {
+    name: 'namespace',
+    alias: 'n',
+    type: String,
+    defaultValue: 'create-block',
+    defaultOption: 'create-block',
+    description: 'The namespace for the block. (default: create-block)',
+  },
+  {
+    name: 'blocksDir',
+    alias: 'b',
+    description: 'The directory where the blocks will be created relative to the current working directory. (default: blocks)',
+    type: String,
+    defaultValue: 'blocks',
+    defaultOption: 'blocks',
+  },
+  {
+    name: 'blockLanguage',
+    alias: 'l',
+    description: 'The language for the block. Accepts `typescript` or `javascript`. (default: typescript)',
+    type: String,
+  },
+  {
+    name: 'help',
+    alias: 'h',
+    description: 'Display this usage guide.',
+    type: Boolean,
+  },
+];
+
+// Get the options from the command line.
+const {
+  namespace,
+  blocksDir: blocksDirectory,
+  blockLanguage,
+  help,
+} : {
+  namespace: string;
+  blocksDir: string;
+  blockLanguage: LanguageType | undefined;
```

Comment:
Yes, that is correct, the command however will not complete as the language relates to block template directory. So by adding `-l foobar` an error will show:
![image](https://github.com/alleyinteractive/alley-scripts/assets/1856961/54d36ffa-b124-4077-a963-b063f2bf7545)

I think in response to your comment what this could use is a more specific error message related to these values before that occurs. That way the user will know that their `-l foobar` is what triggered the failure.

I can add that here.